### PR TITLE
Treat locale case insensitively

### PR DESCRIPTION
**Pull request by diskdance**
BCP 47 tags should be case insensitive. In this patch, it works by converting all locale tags to lowercase.

I don't know if [jquery.i18n](https://github.com/wikimedia/jquery.i18n) is affected. If so, it needs a PR too.

Phab ticket: https://phabricator.wikimedia.org/T359822

### CODE REVIEW

**Review by siddharthvp — CHANGES_REQUESTED** [review]
index.js setLocale() needs to be fixed as well. 

Also, it's probably better to do it directly in BananaEmitter and BananaParser's constructors. Although those classes are supposed to be internal, they are still exported.

I think it would be good to abstract the logic in a normalizeLocale() method and use that everywhere, to avoid the awkwardness of the typeof checks in multiple places.

**Comment by diskdance** [comment]
Fixed. Could you please check again?

**Comment by kartikm** [comment]
I rebased the change to reflect the latest dependency updates, but CI seems failing, will check later tomorrow.

File: src/emitter.js [hunk 1]
```diff
@@ -22,7 +22,7 @@ const strongDirRegExp = new RegExp(
 
 class BananaEmitter {
   constructor (locale) {
-    this.locale = locale
+    this.locale = normalizeLocale(locale)
```
Comment (stephanebisson) [inline src/emitter.js:25]: Here the normalized version of `locale` is assigned to `this.locale` but on the following line, the not-normalized version of `locale` is used.

Comment (diskdance) [reply src/emitter.js:25]: Fixed.

**Comment by diskdance** [comment]
Hello. Any updates on this? Thank you!

**Review by stephanebisson — APPROVED** [review]
It's unclear to me why the `normalizeLocale` function is in `emitter.js` instead of some utils or something explicit like `src/normalizeLocale.js` but looks good otherwise.

**Comment by diskdance** [comment]
Thank you for the approval. Is this ready to merge?

**Review by Abijeet — APPROVED** [review]

**Pull request merged**